SharedSource/UnitTests/FunctionalProgramming/EnumerableExtensionsTest.cs

```diff
@@ -198,7 +228,7 @@ public void CreateSequence_WhilePredicateEvaluatesTrue_WithSingleElement ()
     {
       var element = new Element (0, null);
 
-      IEnumerable<Element> actual = element.CreateSequence (e => e.Parent, e => e != null);
+      IEnumerable<Element> actual = element.CreateSequence (e => e!.Parent!, e => e != null);

```

Comment:
There should be no NFO here needed. That's public, this needs to be clean.

Reply:
If we supply any other predicate this code would produce a NullRefException. Thats why we have an explicit overload that only works with class and can verify that those null values wont make to the nextElementSelector. 
Its definitely not ideal, but i think its reasonable here that you have to tell the compiler that you check for null manually.
(the first ! is not needed and can be removed)

Reply:
ah, I .c. This null-check-predicate option is so stupid anyway. Getting a headache...

Of course, now I'm surprised that we can skip the FIRST NFO, instead of the second. The first one's not null only thanks to the predicate, but that we don't know on a compiler level. the second one can actually be null. I'm starting to wonder if this method can actually be annotated for nulalblity beyond the fact that the sequence will never be null.

Reply:
`e` is not nullable because we specify (implicitly) that the sequence is of not nullable elements, and e is always an element of the sequence